framework/include/auxkernels/ArrayVariableComponent.h

```diff
@@ -0,0 +1,31 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+class ArrayVariableComponent;
+
+template <>
+InputParameters validParams<ArrayVariableComponent>();
+
+class ArrayVariableComponent : public AuxKernel
+{
+public:
+  ArrayVariableComponent(const InputParameters & parameters);
+
+protected:
+  virtual Real computeValue() override;
+
+  /// Holds the solution at current quadrature points
+  const ArrayVariableValue & _u;
+  /// The component
+  const unsigned int _component;
```

Comment:
this would have been better named as `_components` or `_num_components`and the corresponding input parameter as well. We uses `component` to denote the index of a single component (I believe that is what caused the pronghorn failure - not the danger of global parameters, but an unfortunately named parameter in this PR).